repose-aggregator/services/atom-feed/impl/src/main/scala/org/openrepose/nodeservice/atomfeed/impl/AtomFeedServiceImpl.scala

```diff
@@ -0,0 +1,228 @@
+
+    configurationService.subscribeTo(
+      DEFAULT_CONFIG,
+      xsdURL,
+      AtomFeedServiceConfigurationListener,
+      classOf[AtomFeedServiceConfigType]
+    )
+    configurationService.subscribeTo(
+      SYSTEM_MODEL_CONFIG,
+      SystemModelConfigurationListener,
+      classOf[SystemModel]
+    )
+  }
+
+  @PreDestroy
+  def destroy(): Unit = {
+    configurationService.unsubscribeFrom(DEFAULT_CONFIG, AtomFeedServiceConfigurationListener)
+    configurationService.unsubscribeFrom(SYSTEM_MODEL_CONFIG, SystemModelConfigurationListener)
+
+    if (running) {
+      actorSystem.shutdown()
+    }
+  }
+
+  override def registerListener(feedId: String, listener: AtomFeedListener): String = {
+    ifRunning {
+      serviceConfig.getFeed.find(feedConfig => feedId.equals(feedConfig.getId)) match {
+        case Some(feedConfig) =>
+          val listenerId = java.util.UUID.randomUUID().toString
+          logger.debug("Registering a listener with id: " + listenerId + " to feed with id: " + feedId)
+
+          val pastListeners = feedListeners.getOrElse(feedId, Map.empty)
+          feedListeners = feedListeners + (feedId -> (pastListeners + (listenerId -> listener)))
+          if (pastListeners.isEmpty) {
+            val schedule = actorSystem.scheduler.schedule(
+              Duration.Zero,
+              feedConfig.getPollingFrequency.seconds,
+              actorSystem.actorOf(FeedReader.props(
+                feedConfig.getUri,
+                (arf: ActorRefFactory) =>
+                  arf.actorOf(Authenticator.props(feedConfig.getAuthentication)),
+                (arf: ActorRefFactory) =>
+                  arf.actorOf(Notifier.props(feedListeners.getOrElse(feedId, Map.empty).values.toSet)),
+                feedConfig.getEntryOrder
+              )),
+              ReadFeed
+            )
+            feedSchedules = feedSchedules + (feedId -> schedule)
+          }
+
+          listenerId
+        case None =>
+          throw new IllegalArgumentException("feedId parameter does not match a configured feed id")
+      }
+    }
+  }
+
+  override def unregisterListener(listenerId: String): Unit = {
+    ifRunning {
+      feedListeners find {
+        case (_, listeners) => listeners.contains(listenerId)
+      } match {
+        case Some((feedId, listeners)) =>
+          logger.debug("Un-registering the listener: " + listenerId)
+
+          val newListeners = listeners filterNot { case (id, listener) => listenerId.equals(id) }
+          feedListeners = feedListeners + (feedId -> newListeners)
+          if (newListeners.isEmpty) {
+            feedSchedules(feedId).cancel()
+          }
+        case None =>
+          logger.debug("Listener not registered: " + listenerId)
+      }
+    }
+  }
+
+  override def isRunning: Boolean = running
+
+  private def startService(): Unit = {
+    synchronized {
+      if (isInitialized && isServiceEnabled && !running) {
```

Comment:
Interesting, this is a case where to me the cases would add clarity, as to what conditions are being looked for